<h1>Исследование надежности заемщиков (скоринг)<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><ul class="toc-item"><li><ul class="toc-item"><li><ul class="toc-item"><li><span><a href="#Описание-проекта" data-toc-modified-id="Описание-проекта-0.0.0.1"><span class="toc-item-num">0.0.0.1&nbsp;&nbsp;</span>Описание проекта</a></span></li><li><span><a href="#Описание-данных" data-toc-modified-id="Описание-данных-0.0.0.2"><span class="toc-item-num">0.0.0.2&nbsp;&nbsp;</span>Описание данных</a></span></li></ul></li></ul></li></ul></li><li><span><a href="#Откроем-таблицу-и-изучим-общую-информацию-о-данных" data-toc-modified-id="Откроем-таблицу-и-изучим-общую-информацию-о-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Откроем таблицу и изучим общую информацию о данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Обработка-аномальных-значений" data-toc-modified-id="Обработка-аномальных-значений-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Обработка аномальных значений</a></span></li><li><span><a href="#Удаление-пропусков" data-toc-modified-id="Удаление-пропусков-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Удаление пропусков</a></span></li><li><span><a href="#Изменение-типов-данных" data-toc-modified-id="Изменение-типов-данных-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Изменение типов данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.6"><span class="toc-item-num">2.6&nbsp;&nbsp;</span>Категоризация данных</a></span></li><li><span><a href="#Исследуем-данные-и-ответим-на-вопросы" data-toc-modified-id="Исследуем-данные-и-ответим-на-вопросы-2.7"><span class="toc-item-num">2.7&nbsp;&nbsp;</span>Исследуем данные и ответим на вопросы</a></span><ul class="toc-item"><li><span><a href="#Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-количеством-детей-и-возвратом-кредита-в-срок?-2.7.1"><span class="toc-item-num">2.7.1&nbsp;&nbsp;</span>Есть ли зависимость между количеством детей и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-2.7.2"><span class="toc-item-num">2.7.2&nbsp;&nbsp;</span>Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-2.7.3"><span class="toc-item-num">2.7.3&nbsp;&nbsp;</span>Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-2.7.4"><span class="toc-item-num">2.7.4&nbsp;&nbsp;</span>Как разные цели кредита влияют на его возврат в срок?</a></span></li><li><span><a href="#Приведите-возможные-причины-появления-пропусков-в-исходных-данных." data-toc-modified-id="Приведите-возможные-причины-появления-пропусков-в-исходных-данных.-2.7.5"><span class="toc-item-num">2.7.5&nbsp;&nbsp;</span>Приведите возможные причины появления пропусков в исходных данных.</a></span></li><li><span><a href="#Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных." data-toc-modified-id="Объясните,-почему-заполнить-пропуски-медианным-значением-—-лучшее-решение-для-количественных-переменных.-2.7.6"><span class="toc-item-num">2.7.6&nbsp;&nbsp;</span>Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.</a></span></li></ul></li></ul></li><li><span><a href="#Общий-вывод." data-toc-modified-id="Общий-вывод.-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Общий вывод.</a></span></li></ul></div>

##### Описание проекта
Заказчик — *кредитный отдел банка. Нужно разобраться, влияет ли семейное положение и количество детей клиента на факт погашения кредита в срок. Входные данные от банка — статистика о платёжеспособности клиентов.*

Результаты исследования будут учтены при построении модели кредитного скоринга — специальной системы, которая оценивает способность потенциального заёмщика вернуть кредит банку.

##### Описание данных
* children — количество детей в семье
* days_employed — общий трудовой стаж в днях
* dob_years — возраст клиента в годах
* education — уровень образования клиента
* education_id — идентификатор уровня образования
* family_status — семейное положение
* family_status_id — идентификатор семейного положения
* gender — пол клиента
* income_type — тип занятости
* debt — имел ли задолженность по возврату кредитов
* total_income — ежемесячный доход
* purpose — цель получения кредита

## Откроем таблицу и изучим общую информацию о данных

Данные из csv-файла сохраним в переменную `data`. Путь к файлу:**

`/datasets/data.csv`

In [1]:
import pandas as pd# импортируем библиотеку pandas

data = pd.read_csv('https://code.s3.yandex.net/datasets/data.csv')#прочитаем csv-файл

In [2]:
display(data.head(5))# выведем первые 20 строчек датафрейма data
display(data.info())# выведем основную информацию о датафрейме с помощью метода info()

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   children          21525 non-null  int64  
 1   days_employed     19351 non-null  float64
 2   dob_years         21525 non-null  int64  
 3   education         21525 non-null  object 
 4   education_id      21525 non-null  int64  
 5   family_status     21525 non-null  object 
 6   family_status_id  21525 non-null  int64  
 7   gender            21525 non-null  object 
 8   income_type       21525 non-null  object 
 9   debt              21525 non-null  int64  
 10  total_income      19351 non-null  float64
 11  purpose           21525 non-null  object 
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


None

## Предобработка данных

### Удаление пропусков

**Задание 4. Выведем количество пропущенных значений для каждого столбца.**

In [3]:
display(data.isna().sum())

#пропуски в процентном соотношении
isna_in_data = pd.DataFrame(round(data.isna().mean()*100,1)).style.background_gradient('coolwarm')
isna_in_data

children               0
days_employed       2174
dob_years              0
education              0
education_id           0
family_status          0
family_status_id       0
gender                 0
income_type            0
debt                   0
total_income        2174
purpose                0
dtype: int64

В двух столбцах есть пропущенные значения. Один из них — `days_employed`. 
Другой столбец с пропущенными значениями — `total_income` — хранит данные о доходах. На сумму дохода сильнее всего влияет тип занятости, поэтому заполнить пропуски в этом столбце нужно медианным значением по каждому типу из столбца income_type. Например, у человека с типом занятости сотрудник пропуск в столбце total_income должен быть заполнен медианным доходом среди всех записей с тем же типом.

In [4]:
for x in data['income_type'].unique():
    data.loc[(data['income_type'] == x) & (data['total_income'].isna()), 'total_income'] = \
    data.loc[(data['income_type'] == x), 'total_income'].median()

### Обработка аномальных значений

В данных могут встречаться артефакты (аномалии) — значения, которые не отражают действительность и появились по какой-то ошибке. Таким артефактом будет отрицательное количество дней трудового стажа в столбце `days_employed`. 

Обработайте значения в этом столбце: замените все отрицательные значения положительными с помощью метода abs().

In [5]:
data.loc[data['days_employed'] < 0, 'days_employed'] = abs(data['days_employed'])

Для каждого типа занятости выведем медианное значение трудового стажа days_employed в днях.

In [6]:
data.groupby('income_type')['days_employed'].median()

income_type
безработный        366413.652744
в декрете            3296.759962
госслужащий          2689.368353
компаньон            1547.382223
пенсионер          365213.306266
предприниматель       520.848083
сотрудник            1574.202821
студент               578.751554
Name: days_employed, dtype: float64

Выведем перечень уникальных значений столбца children.

In [7]:
data['children'].unique()

array([ 1,  0,  3,  2, -1,  4, 20,  5], dtype=int64)

В столбце `children` есть два аномальных значения. Удалим строки, в которых встречаются такие аномальные значения из датафрейма data.

In [8]:
data = data.loc[(data['children'] != 20) & (data['children'] != -1)]

Выведем перечень уникальных значений столбца `children`, чтобы убедиться, что артефакты удалены.

In [9]:
data['children'].unique()

array([1, 0, 3, 2, 4, 5], dtype=int64)

### Удаление пропусков

Заполним пропуски в столбце days_employed медианными значениями по каждому типу занятости income_type.

In [10]:
for i in data['income_type'].unique():
    data.loc[(data['income_type'] == i) & (data['days_employed'].isna())\
             , 'days_employed'] = data.loc[data['income_type'] == i, 'days_employed'].median()
    

In [11]:
# Провека
data.isna().sum()

children            0
days_employed       0
dob_years           0
education           0
education_id        0
family_status       0
family_status_id    0
gender              0
income_type         0
debt                0
total_income        0
purpose             0
dtype: int64

### Изменение типов данных

**Заменим вещественный тип данных в столбце `total_income` на целочисленный с помощью метода `astype()`.**

In [12]:
data['total_income'] = data['total_income'].astype('int')

### Обработка дубликатов

Выведем на экран количество строк-дубликатов в данных. Если такие строки присутствуют, удалим их.

In [13]:
data['education'] = data['education'].str.lower()# делаем нзавания стобцов в нижнем регистре


Обработаем неявные дубликаты в столбце education

In [14]:
data.duplicated()
data.drop_duplicates().reset_index(drop=True)

children  days_employed  dob_years education  education_id  \
0             1    8437.673028         42    высшее             0   
1             1    4024.803754         36   среднее             1   
2             0    5623.422610         33   среднее             1   
3             3    4124.747207         32   среднее             1   
4             0  340266.072047         53   среднее             1   
...         ...            ...        ...       ...           ...   
21326         1    4529.316663         43   среднее             1   
21327         0  343937.404131         67   среднее             1   
21328         1    2113.346888         38   среднее             1   
21329         3    3112.481705         38   среднее             1   
21330         2    1984.507589         40   среднее             1   

          family_status  family_status_id gender income_type  debt  \
0       женат / замужем                 0      F   сотрудник     0   
1       женат / замужем                 0      F   сотрудник     0   
2       женат / замужем                 0      M   сотрудник     0   
3       женат / замужем                 0      M   сотрудник     0   
4      гражданский брак                 1      F   пенсионер     0   
...                 ...               ...    ...         ...   ...   
21326  гражданский брак                 1      F   компаньон     0   
21327   женат / замужем                 0      F   пенсионер     0   
21328  гражданский брак                 1      M   сотрудник     1   
21329   женат / замужем                 0      M   сотрудник     1   
21330   женат / замужем                 0      F   сотрудник     0   

       total_income                       purpose  
0            253875                 покупка жилья  
1            112080       приобретение автомобиля  
2            145885                 покупка жилья  
3            267628    дополнительное образование  
4            158616               сыграть свадьбу  
...             ...                           ...  
21326        224791             операции с жильем  
21327        155999          сделка с автомобилем  
21328         89672                  недвижимость  
21329        244093  на покупку своего автомобиля  
21330         82047         на покупку автомобиля  

[21331 rows x 12 columns]

### Категоризация данных

На основании диапазонов, указанных ниже, создим в датафрейме `data` столбец `total_income_category` с категориями:**

- 0–30000 — `'E'`;
- 30001–50000 — `'D'`;
- 50001–200000 — `'C'`;
- 200001–1000000 — `'B'`;
- 1000001 и выше — `'A'`.


**Например, кредитополучатель с доходом 25000 назначить категорию `'E'`, а клиенту, получающему 235000, — `'B'`. Используем собственную функцию с именем `categorize_income()` и метод `apply()`.**

In [15]:
def categorize_income(total_income):
    if total_income <= 30000:
        return 'E'
    if total_income <= 50000:
        return 'D'
    if total_income <= 200000:
        return 'C'
    if total_income <= 1000000:
        return 'B'
    return 'A'


In [16]:
data['total_income_category'] = data['total_income'].apply(categorize_income)
data['total_income_category']#проверим

0        B
1        C
2        C
3        B
4        C
        ..
21520    B
21521    C
21522    C
21523    B
21524    C
Name: total_income_category, Length: 21402, dtype: object

**Выведем на экран перечень уникальных целей взятия кредита из столбца `purpose`.**

In [17]:
purpose_unique = data['purpose'].unique()
purpose_unique

array(['покупка жилья', 'приобретение автомобиля',
       'дополнительное образование', 'сыграть свадьбу',
       'операции с жильем', 'образование', 'на проведение свадьбы',
       'покупка жилья для семьи', 'покупка недвижимости',
       'покупка коммерческой недвижимости', 'покупка жилой недвижимости',
       'строительство собственной недвижимости', 'недвижимость',
       'строительство недвижимости', 'на покупку подержанного автомобиля',
       'на покупку своего автомобиля',
       'операции с коммерческой недвижимостью',
       'строительство жилой недвижимости', 'жилье',
       'операции со своей недвижимостью', 'автомобили',
       'заняться образованием', 'сделка с подержанным автомобилем',
       'получение образования', 'автомобиль', 'свадьба',
       'получение дополнительного образования', 'покупка своего жилья',
       'операции с недвижимостью', 'получение высшего образования',
       'свой автомобиль', 'сделка с автомобилем',
       'профильное образование', 'высшее об

Создадим функцию, которая на основании данных из столбца `purpose` сформирует новый столбец `purpose_category`, в который войдут следующие категории:**

- `'операции с автомобилем'`,
- `'операции с недвижимостью'`,
- `'проведение свадьбы'`,
- `'получение образования'`.

**Например, если в столбце `purpose` находится подстрока `'на покупку автомобиля'`, то в столбце `purpose_category` должна появиться строка `'операции с автомобилем'`.**

**Используем собственную функцию с именем `categorize_purpose()` и метод `apply()`. Изучите данные в столбце `purpose` и определите, какие подстроки помогут вам правильно определить категорию.**

In [18]:
def categorize_purpose(purpose):
    if "авто" in purpose:
        return 'операции с автомобилем'
    if 'недв' in purpose or 'жил' in purpose:
        return 'операции с недвижимостью'
    if 'свадь' in purpose:
        return 'проведение свадьбы'
    return 'получение образования'
data['purpose_category'] = data['purpose'].apply(categorize_purpose)# примените функцию методом apply()
data.head()    

children  days_employed  dob_years education  education_id  \
0         1    8437.673028         42    высшее             0   
1         1    4024.803754         36   среднее             1   
2         0    5623.422610         33   среднее             1   
3         3    4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   

                      purpose total_income_category          purpose_category  
0               покупка жилья                     B  операции с недвижимостью  
1     приобретение автомобиля                     C    операции с автомобилем  
2               покупка жилья                     C  операции с недвижимостью  
3  дополнительное образование                     B     получение образования  
4             сыграть свадьбу                     C        проведение свадьбы

### Исследуем данные и ответим на вопросы

#### Есть ли зависимость между количеством детей и возвратом кредита в срок?

In [19]:
# применим группировку по столбцу children и метод объединения функции agg
dependence_children = data.groupby('children').agg({'debt':['count','sum']})

# вывод общего "показательного" процента возвратности
dependence_children_result = (dependence_children['debt']['sum'].sum() / dependence_children['debt']['count'].sum()) 
# вывод процента по группам детей 
dependence_children['returnability'] = (dependence_children['debt']['sum'] / dependence_children['debt']['count']) * 100  
#убираем лишнее для красоты ревьюерам =)
dependence_children = dependence_children.reset_index() 

#выводим результаты
print(f'Общий процент по "возвратности" кредита среди исследуемых групп равен {dependence_children_result:.1%}') 
dependence_children


Общий процент по "возвратности" кредита среди исследуемых групп равен 8.1%


children   debt       returnability
            count   sum              
0        0  14149  1063      7.512898
1        1   4818   444      9.215442
2        2   2055   194      9.440389
3        3    330    27      8.181818
4        4     41     4      9.756098
5        5      9     0      0.000000

**Вывод:**  Категория людей "без детей" (`7.5%`) выплачивают кредиты быстрее всех других категорий, что в нашем случае является идеальным образом заемщика (для выдачи кредитов в будующем), чем категория людей с 1-им 2-мя детьми.

#### Есть ли зависимость между семейным положением и возвратом кредита в срок?

In [20]:
#применим группировку по столбцу family_status и в функции вычислим сумму и колличество
dependence_family_status = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = ['sum', 'count']) 

#вычислим отношение суммы к количсетву и найдем процент людей по каждой группе, кто не выплатил кредит
dependence_family_status['returnability'] = (dependence_family_status['sum']['debt'] / dependence_family_status['count']['debt'])*100

#найдем общий процент должников по всем группам - он не изменился =)
dependence_family_status_result = (dependence_family_status['sum']['debt'].sum() / dependence_family_status['count']['debt'].sum())

#выводим результаты
print(f'Общий процент по "возвратности" кредита среди исследуемых групп равен {dependence_family_status_result:.1%}') 
dependence_family_status


Общий процент по "возвратности" кредита среди исследуемых групп равен 8.1%


sum  count returnability
                      debt   debt              
family_status                                  
Не женат / не замужем  273   2799      9.753483
в разводе               84   1189      7.064760
вдовец / вдова          63    952      6.617647
гражданский брак       385   4160      9.254808
женат / замужем        927  12302      7.535360

**Вывод:** Самый маленький процент - у вдов и вдовцов , они вовремя гасят кредиты, чем все остальные. Потом идут пары в разводе у них `7%`. Не торопятся не женатые и не замужние - почти `10%`. 

#### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

In [21]:
#применим группировку по столбцу total_income_category и в функции вычислим сумму и колличество
dependence_total_income_category = data.groupby('total_income_category').agg({'debt':['count','sum']})

#вычислим отношение суммы к количсетву и найдем процент людей по каждой группе, кто не выплатил кредит
dependence_total_income_category['returnability'] = (dependence_total_income_category['debt']['sum'] / dependence_total_income_category['debt']['count']) * 100
#выводим результаты
dependence_total_income_category

debt       returnability
                       count   sum              
total_income_category                           
A                         25     2      8.000000
B                       5014   354      7.060231
C                      15992  1353      8.460480
D                        349    21      6.017192
E                         22     2      9.090909

**Вывод:** Обратим внимание на то, что выборки крайне несбалансированны и для выявления зависимости нужно сравнивать только самые многочисленные выборки B и C. Опираясь на данные , полученные в таблице, можно утверждать `категория C` имеет один из самых высоких показателей задолженности. А `категория В` только 7.06. 

Вывод группа людей, с самым высоким доходом являются большими должниками.

#### Как разные цели кредита влияют на его возврат в срок?

In [22]:
#применим группировку по столбцу purpose_category и в функции вычислим сумму и колличество
dependence_purpose_category = data.groupby('purpose_category').agg({'debt':['count','sum']})

#вычислим отношение суммы к количсетву и найдем процент людей по каждой группе, кто не выплатил кредит
dependence_purpose_category['returnability'] = (dependence_purpose_category['debt']['sum'] / dependence_purpose_category['debt']['count']) * 100

#выводим результаты
dependence_purpose_category

debt      returnability
                          count  sum              
purpose_category                                  
операции с автомобилем     4288  400      9.328358
операции с недвижимостью  10780  780      7.235622
получение образования      3997  369      9.231924
проведение свадьбы         2337  183      7.830552

**Вывод:** Почти на одном уровне категрии людей с целью кредита: операции с автомобилем и получение образования - они выплачивают кредит медленней, чем категория людей с целью: операции с недвижимостью и проведение свадьбы. Быстрее всего закрывают ипотеку.

#### Приведите возможные причины появления пропусков в исходных данных.

*Ответ:* По итогам анализа датафрейма (а именно - установление закономерности в пропусках) было выявлено: **пропуски вызваны человеческим фактором. Закономерностей в пропусках найдено не было.**

Методы исследования датафрейма на появление пропусков:
* Соотношение пропусков в столбце days_employed с пустыми значениями с уникальными значниями столбца income_type `data.loc[data['days_employed'].isna(), 'income_type'].unique()`
* Соотношение пропусков в столбце days_employed с пустыми значениями с уникальными значниями столбца gender 
`data.loc[data['days_employed'].isna(), 'gender'].unique()`

#### Объясните, почему заполнить пропуски медианным значением — лучшее решение для количественных переменных.

*Ответ:* Пропуски в количественных переменных лучше заполнять медианным значением, потому что , среднее значение некорректно характеризует данные, когда некоторые значения сильно выделяются среди большинства. Например, люди , которые получают заработную плату категории Е (до 30000), всего 22 человека из ~21000, а Категории А (до 1000001), вычисление среднего значения в данном случае будет некоректно. 

## Общий вывод.

**Из вводных - мы получили датафрейм с данными о заемщиках. Гипотезы, которые были выстроены, в большинсве своем, не подтвердились, так как разница между показателями в 1 - 1.5% не является существенной (относительно среднего), но если углубиться в каждую гипотезу, мы можем найти некоторые интересные тонкости:** 

* Зависимость между колчеством детей и возратом кредита в срок подтвердилась `(разница 2%)` относительно групп, но не общего процента. Общий процент от "возвратности" кредита среди групп с разным количеством детей равен 8.1%. Минимальне значение в 7.5% группа без детей. Максимальное значение в 9.7% группа с 4-мя детьми. Быстре выплачивают кредиты группа без детей. 

* Зависимость между семейным положением и возвратом кредита в срок подтвердилась `(разница 2%)` относительно групп, но не общего процента. Группа людей, не имеющих "долговых обязательств" перед друг другом (Не женат / не замужем) - 9.7%, менее организованы , чем группа людей "вдовец / вдова" 6.6%.

* Зависимость между уровнем дохода и возвратом кредита в срок срок подтвердилась `(разница 3%)`. Группа людей с категорией D (с заработком 30001–50000) выплачивают бысрее кредит, чем все остальные, даже категории А (с заработком 1000001 и выше), где количество дебиторов одинаковое.

* Гипотеза о зависимости целей на возврат, тоже подтвердились "операции с недвижимостью" 7.2% `(разница 2%)`, следовательно кредиты на недвижимость выплачивают быстрее , чем "операции с автомобилем" 9.3. 